File: Contests/2019-2020/Bronze/Feb/swap.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(2)
using namespace std;
typedef long long ll;
const int MAXN = 105;

int n, k, a1, a2, b1, b2, a[105], b[105];

int solve(int p) {
    int cur = p, loop = 0;
    while (1) {
        if (cur>=a1 && cur<=a2) cur = (a1+a2) - cur;
        if (cur>=b1 && cur<=b2) cur = (b1+b2) - cur;
        ++loop;
        if (cur == p) break;
    }
    for (int i = 0; i < k%loop; ++i) {
        if (cur>=a1 && cur<=a2) cur = (a1+a2) - cur;
        if (cur>=b1 && cur<=b2) cur = (b1+b2) - cur;
    }
    return cur;
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("swap.in", "r", stdin);
    freopen("swap.out", "w", stdout);
    cin >> n >> k >> a1 >> a2 >> b1 >> b2;
    for (int i = 1; i <= 100; ++i) a[i] = i;
    for (int i = 1; i <= n; ++i) b[solve(i)] = i;
    for (int i = 1; i <= n; ++i) cout << b[i] << endl;
    return 0;
}

File: Contests/2019-2020/Bronze/Feb/swap.py
import sys
sys.stdin = open("swap.in", "r")
sys.stdout = open("swap.out", "w")

n, k = map(int, input().split())
a1, a2 = map(int, input().split())
b1, b2 = map(int, input().split())
a = [i for i in range(n+1)]
b = [0] * (n+1)

def solve(p):
    loop = 0
    cur = p
    while 1:
        if a1<=cur<=a2:
            cur = (a1+a2) - cur
        if b1<=cur<=b2:
            cur = (b1+b2) - cur
        loop += 1
        if cur==p: break
    for i in range(k % loop):
        if a1<=cur<=a2:
            cur = (a1+a2) - cur
        if b1<=cur<=b2:
            cur = (b1+b2) - cur
    return cur


for i in range(1, n+1):
    b[solve(i)] = i
for i in range(1, n+1):
    print(b[i])